File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Cardlytics/CardlyticsSettings.cs
using System.Collections.Generic;

namespace Psi.Data.Models.ClientConfigurationModels.Cardlytics
{
    public class CardlyticsSettings : SettingsBaseHelper
    {
        public CardlyticsSettings(ISettingsBase settingsBase): base(settingsBase)
        {
        }

        [SettingKey("X.App.HomeBanking.CardlyticsEnabled")]
        public bool CardlyticsEnabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HBAdminBOL.CardlyticsEnabled")]
        public bool AdminCardlyticsEnabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.CardlyticsAccountTypes")]
        public List<string> CardlyticsAccountTypes
        {
            get => GetListValue();
            set => SetValue(value);
        }

        /// <summary>
        /// If true, then show the 'update redemption suffix' row in the get Member view in Admin
        /// </summary>
        [SettingKey("Cardlytics.Admin.ShowSuffixUpdateRowInAdmin")]
        public bool ShouldShowSuffixUpdateRowInAdmin
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        /// <summary>
        /// Specifies the version of jQuery to use with the Cardlytics provided javascript.
        /// </summary>
        [SettingKey("Cardlytics.jQuery.Version")]
        public string JQueryVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Cardlytics.DomInjectionVersion.Enabled")]
        public bool DomInjectionVersionEnabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("Cardlytics.DomInjectionVersion.MainScriptName")]
        public string MainScriptName
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Cardlytics.DomInjectionVersion.ConfigScriptName")]
        public string ConfigScriptName
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Cardlytics.DomInjectionVersion.InitScriptName")]
        public string InitScriptName
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Cardlytics.DomInjectionVersion.FiFolder")]
        public string FiFolder
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Cardlytics.DomInjectionVersion.OpsServerUrl")]
        public string OpsServerUrl
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Cardlytics.DomInjectionVersion.JQueryVersion")]
        public string DomInjectionVersionJQueryVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Cardlytics.DomInjectionVersion.VueVersion")]
        public string VueVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Cardlytics/CardlyticsSettings.ts
// Generated imports
import { CardlyticsAccountTypes } from '../CardlyticsAccountTypes';

export interface CardlyticsSettings {
    /** @settingKey X.App.HomeBanking.CardlyticsEnabled */
    cardlyticsEnabled: boolean;
    /** @settingKey X.App.HBAdminBOL.CardlyticsEnabled */
    adminCardlyticsEnabled: boolean;
    /** @settingKey X.App.HomeBanking.CardlyticsAccountTypes */
    list: CardlyticsAccountTypes;
    /** @settingKey Cardlytics.Admin.ShowSuffixUpdateRowInAdmin */
    /**
     * /// <summary>
     * /// /// <summary>
     * /// /// If true, then show the 'update redemption suffix' row in the get Member view in Admin
     * /// /// </summary>
     * /// </summary>
     */
    shouldShowSuffixUpdateRowInAdmin: boolean;
    /** @settingKey Cardlytics.jQuery.Version */
    /**
     * /// <summary>
     * /// /// <summary>
     * /// /// Specifies the version of jQuery to use with the Cardlytics provided javascript.
     * /// /// </summary>
     * /// </summary>
     */
    jQueryVersion: string;
    /** @settingKey Cardlytics.DomInjectionVersion.Enabled */
    domInjectionVersionEnabled: boolean;
    /** @settingKey Cardlytics.DomInjectionVersion.MainScriptName */
    mainScriptName: string;
    /** @settingKey Cardlytics.DomInjectionVersion.ConfigScriptName */
    configScriptName: string;
    /** @settingKey Cardlytics.DomInjectionVersion.InitScriptName */
    initScriptName: string;
    /** @settingKey Cardlytics.DomInjectionVersion.FiFolder */
    fiFolder: string;
    /** @settingKey Cardlytics.DomInjectionVersion.OpsServerUrl */
    opsServerUrl: string;
    /** @settingKey Cardlytics.DomInjectionVersion.JQueryVersion */
    domInjectionVersionJQueryVersion: string;
    /** @settingKey Cardlytics.DomInjectionVersion.VueVersion */
    vueVersion: string;
}
